overlord/snapstate/snapmgr.go

```diff
@@ -551,6 +556,45 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// ensureSnapConfineApparmor ensures that we have a valid apparmor
+// profile for snap-confine when we are in re-exec mode
+func (m *SnapManager) ensureSnapConfineApparmor() error {
```

Comment:
This is fixed now, as @jdstrand suggested the code will only keep the "current" profile around and loaded and will discard older profiles.